core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sequencediag/ScenarioService.java

```diff
@@ -88,6 +86,7 @@
 import org.polarsys.capella.core.model.helpers.queries.QueryIdentifierConstants;
 import org.polarsys.capella.core.model.helpers.queries.filters.OnlySharedDataOrEventOrUnsetFilter;
 import org.polarsys.capella.core.sirius.analysis.CapellaServices;
+import org.polarsys.capella.core.sirius.analysis.EventContextServices;
```

Comment:
**Preambule:**
Thank you for the contribution. The new caching system seems cool and I can see the benefits from this change :+1: 

My comments ones are mostly related to style, class location, variable naming conventions and are not all that functional except for a few. Don't hesitate to come back to me if you need more details.